fdbserver/TLogGroup.h

```diff
@@ -0,0 +1,168 @@
+/*
+ * TLogGroup.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FDBSERVER_TLOGGROUP_H
+#define FDBSERVER_TLOGGROUP_H
+#include "fdbclient/CommitProxyInterface.h"
+#include "fdbclient/NativeAPI.actor.h"
+#include "fdbserver/IKeyValueStore.h"
+#pragma once
+
+#include "fdbrpc/Locality.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "flow/FastRef.h"
+#include "flow/IRandom.h"
+#include "flow/network.h"
+#include <unordered_map>
+#include <vector>
+
+// TODO: Monitor the groups, and if new tLogs need to added/removed, as workers are removed/added.
+// TODO: Add unit-test for serialization.
+// TODO:
+
+struct TLogWorkerData;
+class TLogGroup;
+class TLogGroupCollection;
+
+typedef Reference<TLogWorkerData> TLogWorkerDataRef;
+typedef Reference<TLogGroup> TLogGroupRef;
+typedef Reference<TLogGroupCollection> TLogGroupCollectionRef;
+
+// `TLogGroupCollection` manages, recruits and tracks all the TLogGroups in the system.
```

Comment:
This is fine for now, i.e., for a single DC. When supporting HA, we'll need to have tlog goups for satellite and remote DC. Are these separate `TLogGroupCollection`s, or are separate fields inside a `TLogGroupCollection`? Maybe you can add some comments/TODOs here.